### ISSUE TITLE
implement draw distance?

### ISSUE THREAD

**Issue by SteigerHSV** [ptitSeb/stuntcarremake#11]
Hey!

I was posting issue in gl4es :D. I do that because i want install and share instrucion with others how to play this game on pocketCHIP 

It works now but when "menu" works fine with like 10-20fps, game itself run too slow. Idk. how you draw track model but maybe its possible to implement draw distance? maybe this could increase it 

i test it with:
```
LD_LIBRARY_PATH=~/gl4es/lib LIBGL_FB=3 ./stuntcarracer
LD_LIBRARY_PATH=~/gl4es/lib ./stuntcarracer
```
and both option with -n

and i don't want to be offence because different machine spec etc. but emulating oryginal game via v64 works faster.

**Comment by ptitSeb** [comment]
The game should not be slow. I play it on a Pandora and it's fast. Be sure to disable msaa with `-nomsaa` and run in fullscreen with desktop resolution with `-d`.
Did you built an SDL2 version of it ?  Also, you built a CHIP version of StuntCar right (with `make CHIP=1`)?

**Comment by SteigerHSV** [comment]
I check specs and Chip is more powerfull than Pandora Classic but kinda similar spec to Rebirth, but this is what i do step by step:

```
sudo apt install \

cmake \
libsdl-mixer1.2-dev \
libsdl-image1.2-dev \
libsdl-ttf2.0-dev \
libopenal-dev \
libglm-dev

sudo apt-get install chip-mali-userspace

git clone https://github.com/ptitSeb/gl4es

cd gl4es

mkdir build; cd build; cmake .. -DCHIP=1 -DCMAKE_BUILD_TYPE=RelWithDebInfo; make

cd ~

git clone https://github.com/ptitSeb/stuntcarremake
cd stuntcarremake
make CHIP=1

and run it via: 
LD_LIBRARY_PATH=~/gl4es/lib LIBGL_FB=3 ./stuntcarremake/stuntcarracer -n
```

video showing command 
`LD_LIBRARY_PATH=~/gl4es/lib LIBGL_FB=1 ./stuntcarremake/stuntcarracer -n -d
`https://www.youtube.com/watch?v=ON6gjfY-Nho

also terminal output via command > output.txt:
[output.txt](https://github.com/ptitSeb/stuntcarremake/files/3999261/output.txt)
and command &> output2.txt:
[output2.txt](https://github.com/ptitSeb/stuntcarremake/files/3999275/output2.txt)

**Comment by ptitSeb** [comment]
I don't think you are using gl4es, it's probaby mesa software that is running.

I guess gl4es libGL.so is not in `~/gl4es/lib`, but maybe in `~ /gl4es/build/lib` instead?

**Comment by SteigerHSV** [comment]
so if it build via your instruction than yes libGL.so.1 is in /build/lib

so i use:
`LD_LIBRARY_PATH=~/gl4es/build/lib LIBGL_FB=1 ./stuntcarremake/stuntcarracer -n -d`
i think its trying do something but no mather if its is with or without LIBGL_FB or is set to 1-3 or -n -d
it works at 0fps. 
[output3.txt](https://github.com/ptitSeb/stuntcarremake/files/4000749/output3.txt)
[output4.txt](https://github.com/ptitSeb/stuntcarremake/files/4000750/output4.txt)

**Comment by ptitSeb** [comment]
Ok, it's gl4es now that is used, and it seems the ARM GLES driver is used too. Still, if it's 0 fps, that means there is an issue with the driver somewhere.
Can you try `es2gears`  to confirm everything is working properly on GLES driver side?

**Comment by SteigerHSV** [comment]
i can try and test what ever you want, just type command to me and i will send you results.

I also create icon for your remake (also for pocket) so if you want feel free to use it

![stuntcarremake](https://user-images.githubusercontent.com/49843660/71486603-7aee0700-2817-11ea-906a-e43243c29a91.png)

**Comment by ptitSeb** [comment]
Oh, nice icon! Yeah, Maybe I'll grab it.

Just type `es2gears` and you should see 2 3D gears turning. The fps should be at 60fps, something like that. That's a simple GLES 2 program used to test is ES is correctly installed

May you'll need to install es2gears if `es2gears` gives `command not found`. It's probably something like `sudo apt install mesa-utils-extra` to have it.

**Comment by SteigerHSV** [comment]
Check that and yes i didint had `es2gears` and when i install it, it works at looks like 60 fps i can even rotate them using arrows.

**Comment by ptitSeb** [comment]
The fps should be printed to the console every second or 2 (I think, not totaly sure)...
If you have 60fps in es2gears, you should have a smooth framerate in stuntcarremake too.
Something is wrong, but I cannot understand what...

Do you have ssh access on you pocketchip? It would be very usefull to do a `lsof | grep stuntcar` while stunt car is running to see loaded module, and also some `sudo perf top` to see what is eating all cpu power.

**Comment by SteigerHSV** [comment]
I check. That es2gear shows 51-58 fps 

I didint hav `lsof` So i use `sudo apt install lsof` I have it now but when i run stunt car i cant even ctrl-tab only i can do is exit via esc

**Comment by ptitSeb** [comment]
Yep, that why I suggested you use ssh...

I assume ssh service is running, so from a windows PC use putty to connect to your PocketCHIP (or simply use ssh from a linux terminal if you have linux pc.

**Comment by SteigerHSV** [comment]
This is geting harder than i was thinking. I can try on ubuntu because i use it to flash chip but i need to know what to do step by step. CHIP should be able SSH after all its developing board similar to other ones
